### DISCUSSION TITLE
Why does testing current version fails under linux?

### DISCUSSION THREAD

**Discussion by octaeder** [texstudio-org/texstudio#2421] [Q&A] [answered]
[Log](https://github.com/texstudio-org/texstudio/runs/6966055866?check_suite_focus=true) for Linux-autoTests shows as a result for my new test:
```
"FAIL!  : VersionTest::currentVersion(currentVersion) '(Version::current().Version::isEmpty())==(valid)' returned FALSE. (equal failed: got \"1\" !=expected \"0\" )"
"   Loc: [.moc/../src/tests/utilsversion_t.h(113)]
```
It fails because the version object created from constant TEXSTUDIO_GIT_REVISION is empty. This doesn't happen with win10. It looks like regular expression ```"^((\\d+)(\\.\\d+)*)([a-zA-Z]+)?(\\d*)?(-(\\d+)-)?($|[^.])"``` doesn't match the constant. It matches strings like version tags (ex: "4.3.0beta1") or git revisions (ex: "4.3.0beta1-24-g5c925a387"), and is used to split the version string into its parts ("4.3.0", "beta", "1", plus "24" in second ex.).

The reg. exp. is based on one that was used before. I added term ```($|[^.])``` at the end. This was done to recognize that ```"4.3.0.beta1"``` is not correct (dot after 0). The first part of regExp matches "4.3.0", but type, revision, and commit parts can't match the dot. And the added part prohibits the dot.

So I assume that constant TEXSTUDIO_GIT_REVISION might be empty. Can you please help me?

Thx.

PS: After adding some debug information to Version::current() the run passes:
```
"QDEBUG : VersionTest::currentVersion(currentVersion) (\"8706144\", \"\", \"\", \"\")"
"QDEBUG : VersionTest::currentVersion(currentVersion) \"4.3.00\""
"QDEBUG : VersionTest::currentVersion(currentVersion) (\"8706144\", \"\", \"\", \"\")"
"QDEBUG : VersionTest::currentVersion(currentVersion) \"4.3.00\""
```
Each call of current() should print a list of parts used to  fill the version object, and the string representation of the object. So, why 4 lines, why do the parts not match the string and why 00?

After adding ```qDebug() << TEXSTUDIO_GIT_REVISION;``` I see:
```
"QDEBUG : VersionTest::currentVersion(currentVersion) e281119"
"QDEBUG : VersionTest::currentVersion(currentVersion) ()"
"QDEBUG : VersionTest::currentVersion(currentVersion) e281119"
"QDEBUG : VersionTest::currentVersion(currentVersion) ()"
```
This would confirme that the constant equals "e281119" and thus version object is empty. Can it be? Looks like some kind of commit number.

**Comment by sunderme** [comment, marked as answer]
the tags are not checked out.
don't rely that git will give a meaningful output on the CI action, just do fully deterministic comparisons.

**Reply by octaeder** [reply]
I changed current() so test can use a fixed parameter:
```
static Version current(const QString &versionString = TEXSTUDIO_GIT_REVISION);
```
and
```
QEQUAL(Version::current("4.3.0beta1-24-g5c925a387").Version::isEmpty(), valid);
```
